internal/helper/yawollet.go

```diff
@@ -788,6 +820,55 @@ func UpdateKeepalivedStatus(
 		"StatsUpToDate", "Keepalived stat file is newer than 5 min")
 }
 
+func UpdateKeepalivedPIDStatus(
+	ctx context.Context,
+	c client.StatusWriter,
+	lbm *yawolv1beta1.LoadBalancerMachine,
+) error {
+	pidFile, err := os.ReadFile("/run/keepalived.pid")
+	if err != nil {
+		return UpdateLBMConditions(ctx, c, lbm,
+			KeepalivedProcess,
+			ConditionFalse,
+			"CouldNotGetPIDFile", "Could not get pid file")
+	}
+	pidFileContent := strings.ReplaceAll(string(pidFile), "\n", "")
+	pidID, err := strconv.ParseInt(pidFileContent, 10, 32)
+	if err != nil {
+		return UpdateLBMConditions(ctx, c, lbm,
+			KeepalivedProcess,
+			ConditionFalse,
+			"CouldNotGetPID", "Could not get pid")
+	}
+
+	keepalivedProc, err := process.NewProcess(int32(pidID))
+	if err != nil {
+		return UpdateLBMConditions(ctx, c, lbm,
+			KeepalivedProcess,
+			ConditionFalse,
+			"CouldNotKeepalivedProcess", "Could not get keepalived process")
+	}
+	keepalivedRunning, err := keepalivedProc.IsRunning()
+	if err != nil {
+		return UpdateLBMConditions(ctx, c, lbm,
+			KeepalivedProcess,
+			ConditionFalse,
+			"CouldNotKeepalivedProcess", "Could not get keepalived process")
```

Comment:
maybe use another message here so we can distinct between the two errors for debugging, but thats just a little thought